hs_core/models.py

```diff
@@ -4032,37 +4032,37 @@ def get_crossref_deposit_xml(self, pretty_print=True):
         logger = logging.getLogger(__name__)
 
         def get_funder_id(funder_name):
-            """Return funder_id for a given funder_name from Crossref funders registry.
-            Crossref API Documentation: https://api.crossref.org/swagger-ui/index.html#/Funders/get_funders
+            """Return funder_uri for a given funder_name from ror funders registry.
+            ROR API Documentation: https://ror.readme.io/docs/api-query
             """
-
-            # url encode the funder name for the query parameter
-            words = funder_name.split()
-            # filter out words that contain the char '.'
-            words = [word for word in words if '.' not in word]
-            encoded_words = [urllib.parse.quote(word) for word in words]
-            # match all words in the funder name
-            query = "+".join(encoded_words)
-            # if we can't find a match in first 50 search records then we are not going to find a match
-            max_record_count = 50
-            email = settings.DEFAULT_DEVELOPER_EMAIL
-            url = f"https://api.crossref.org/funders?query={query}&rows={max_record_count}&mailto={email}"
+            # # url encode the funder name for the query parameter
+            # words = funder_name.split()
+            # # filter out words that contain the char '.'
+            # words = [word for word in words if '.' not in word]
+            # encoded_words = [urllib.parse.quote(word) for word in words]
+            # # match all words in the funder name
+            # query = "+".join(encoded_words)
+            url = "https://api.ror.org/v2/organizations?filter=types:funder"
             funder_name = funder_name.lower()
             response = requests.get(url, verify=False)
             if response.status_code == 200:
                 response_json = response.json()
-                if response_json['status'] == 'ok':
-                    items = response_json['message']['items']
-                    for item in items:
-                        if item['name'].lower() == funder_name:
-                            return item['uri']
-                        for alt_name in item['alt-names']:
-                            if alt_name.lower() == funder_name:
-                                return item['uri']
-                    return ''
+                items = response_json['items']
+                for item in items:
```

Comment:
I like your updates to use the ROR query. I haven't tested it, but it looks reasonable. I'm assuming you tested it against some edge cases. Ideally you can check those edge case tests into version control.

But the get_funder_id function still has the issue that it returns the url of the organization (aka the `item['id']` instead of returning the ROR equivalent of a DOI.